File: app/src/main/java/com/coinninja/coinkeeper/cn/wallet/service/CNWalletBinder.java
package com.coinninja.coinkeeper.cn.wallet.service;

import android.os.Binder;

import com.coinninja.coinkeeper.cn.wallet.interfaces.CNWalletServicesInterface;

public class CNWalletBinder extends Binder {
    private final CNWalletServicesInterface cnWalletService;

    public CNWalletBinder(CNWalletServicesInterface cnWalletService) {
        this.cnWalletService = cnWalletService;
    }

    public CNWalletServicesInterface getService() {
        return cnWalletService;
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/cn/wallet/service/CNWalletBinder.kt
package com.coinninja.coinkeeper.cn.wallet.service

import android.os.Binder
import app.dropbit.annotations.Mockable

import com.coinninja.coinkeeper.cn.wallet.interfaces.CNWalletServicesInterface

@Mockable
class CNWalletBinder(val service: CNWalletServicesInterface) : Binder()
